### ISSUE TITLE
Internal error, maybe related to virtual libraries

### ISSUE THREAD

**Issue by psteckler** [ocaml/dune#2980]
I started to add a virtual library to a project, and running `dune build` gives:
```
Internal error, please report upstream including the contents of _build/log.
Description:
(Option.value_exn)
Backtrace:
Raised at file "src/stdune/exn.ml", line 40, characters 5-10
Called from file "src/stdune/option.ml" (inlined), line 40, characters 12-48
Called from file "src/lib.ml", line 1168, characters 36-57
Called from file "src/lib.ml", line 625, characters 21-30
```
I see two log files:
```
# dune build config.mlh
# OCAMLPARAM: unset
$ /home/steck/.opam/4.07.1/bin/ocamlc.opt -config > /tmp/dune7aad74.output
$ (cd lib/coda_digestif && /home/steck/.opam/4.07.1/bin/ocaml -I +compiler-libs /home/steck/CodaProtocol/0/coda/src/_build/.dune/default/lib/coda_digestif/dune.ml)
$ (cd external/ppx_optcomp/test && /home/steck/.opam/4.07.1/bin/ocaml -I +compiler-libs /home/steck/CodaProtocol/0/coda/src/_build/.dune/default/external/ppx_optcomp/test/dune.ml)
$ (cd config && /home/steck/.opam/4.07.1/bin/ocaml -I +compiler-libs /home/steck/CodaProtocol/0/coda/src/_build/.dune/default/config/dune.ml)
```
and
```
# dune b config.mlh
# OCAMLPARAM: unset
$ /usr/bin/nproc > /tmp/dune7b29a8.output 2> /dev/null
$ /home/steck/.opam/4.07.1/bin/ocamlc.opt -config > /tmp/dune6a71a3.output
$ (cd src/external/ppx_optcomp/test && /home/steck/.opam/4.07.1/bin/ocaml -I +compiler-libs /home/steck/CodaProtocol/0/coda/_build/.dune/default/src/external/ppx_optcomp/test/dune.ml)
$ (cd src/config && /home/steck/.opam/4.07.1/bin/ocaml -I +compiler-libs /home/steck/CodaProtocol/0/coda/_build/.dune/default/src/config/dune.ml)
$ (cd src/lib/coda_digestif && /home/steck/.opam/4.07.1/bin/ocaml -I +compiler-libs /home/steck/CodaProtocol/0/coda/_build/.dune/default/src/lib/coda_digestif/dune.ml)
```
This particular build target has nothing to do with the virtual library, but maybe dune is nonetheles scanning the dune file for that library, resulting in the error.

**Comment by psteckler** [comment]
This is dune 1.9.1, BTW.

**Comment by psteckler** [comment]
apparently fixed in 1.11.4
